ql/src/test/queries/clientpositive/fetch_deleted_rows.q

```diff
@@ -0,0 +1,48 @@
+set hive.support.concurrency=true;
+set hive.txn.manager=org.apache.hadoop.hive.ql.lockmgr.DbTxnManager;
+SET hive.vectorized.execution.enabled=false;
+
+create table t1(a int, b varchar(128)) stored as orc tblproperties ('transactional'='true');
+
+insert into t1(a,b) values (1, 'one'), (2, 'two');
+
+delete from t1 where a = 1;
+
+insert into t1(a,b) values (3, 'three'), (4, 'four'), (4, 'four again'), (5, 'five');
+
+alter table t1 set tblproperties ('acid.fetch.deleted.rows'='true');
+select t1.ROW__IS__DELETED, * from t1;
```

Comment:
Just leaving a note. I think this syntax passing the table properties in the query should also work (if that's the case, maybe we can add a query with it):
```
select t1.ROW__IS__DELETED, *
from t1('acid.fetch.deleted.rows'='true');
```

Reply:
When the logical plan is built from the AST and `HiveTableScans` are instantiated table properties defined in the AST are lost.
When the optimized logical plan is converted back to AST only properties coming from Druid and JDBC scans are populated. We could add additional properties from `HiveTableScans` in the same way.

I think passing `acid.fetch.deleted.rows` as a table property in the query is a good idea and we can benefit from it  when building an Incremental MV rebuild plan. I will try to include this into [HIVE-24854](https://issues.apache.org/jira/browse/HIVE-24854)